solutions/cses-1084.mdx

```diff
@@ -2,44 +2,70 @@
 id: cses-1084
 source: CSES
 title: Apartments
-author: Danh Ta Chi Thanh
+author: Nathan Gong, Danh Ta Chi Thanh
 ---
 
-## Solution
+[Unofficial Editorial (C++)](https://codeforces.com/blog/entry/83295)
 
-[Unofficial Editorial](https://codeforces.com/blog/entry/83295)
+We can use a greedy approach to solve this problem. First, sort the applicants
+and apartments. We can keep two pointers $i$ and $j$ (initialized to 0),
+which keep track of the current index of the applicant and apartment we are
+looking at respectively. Then, while there are more applicants and apartments to
+look through, we repeatedly run the following check:
+
+- If $|\texttt{applicants}[i] - \texttt{apartments}[j]| \leq k$, we have found a
+  suitable apartment for the current applicant. Thus, we increment $i$, $j$, and
+  our answer.
+- Otherwise, $\texttt{apartments}[j]$ is either too big or too small for
+  $\texttt{applicants}[i]$. We can increment $i$ or $j$ accordingly.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(n\log(n)+m\log(m))$
+
+<LanguageSection>
+
+<CPPSection>
 
 ```cpp
 #include <bits/stdc++.h>
 
 using namespace std;
 
-//constant initialization
-const int maxn=2e5+10;
+// Constant initialization
```

Comment:
not necessary?

Reply:
not sure what you mean. howcome

Reply:
the comment- i mean it's pretty clear this is initializing a constant